plugin/oauth2/src/OAuth2.php

```diff
@@ -0,0 +1,382 @@
+
+                self::SETTING_RESPONSE_RESOURCE_OWNER_URLS => 'text',
+
+                self::SETTING_LOGOUT_URL => 'text',
+
+                self::SETTING_BLOCK_NAME => 'text',
+
+                self::SETTING_MANAGEMENT_LOGIN_ENABLE => 'boolean',
+                self::SETTING_MANAGEMENT_LOGIN_NAME => 'text',
+            ]
+        );
+    }
+
+    /**
+     * Instance the plugin.
+     *
+     * @staticvar null $result
+     *
+     * @return $this
+     */
+    public static function create()
+    {
+        static $result = null;
+
+        return $result ? $result : $result = new self();
+    }
+
+    /**
+     * @return GenericProvider
+     */
+    public function getProvider()
+    {
+        return new GenericProvider(
+            [
+                'clientId' => $this->get(self::SETTING_CLIENT_ID),
+                'clientSecret' => $this->get(self::SETTING_CLIENT_SECRET),
+                'redirectUri' => api_get_path(WEB_PLUGIN_PATH).'oauth2/src/callback.php',
+
+                'urlAuthorize' => $this->get(self::SETTING_AUTHORIZE_URL),
+                // 'scopes' => $this->get(self::SETTING_SCOPES) or null,
+                // 'scopeSeparator' => $this->get(self::SETTING_SCOPE_SEPARATOR) ?: ',',
+
+                'urlAccessToken' => $this->get(self::SETTING_ACCESS_TOKEN_URL),
+                'accessTokenMethod' => $this->get(self::SETTING_ACCESS_TOKEN_METHOD) ?: GenericProvider::METHOD_POST,
+                //'accessTokenResourceOwnerId' => $this->get(self::SETTING_ACCESS_TOKEN_RESOURCE_OWNER_ID)
+                //    ?: GenericProvider::ACCESS_TOKEN_RESOURCE_OWNER_ID,
+
+                'urlResourceOwnerDetails' => $this->get(self::SETTING_RESOURCE_OWNER_DETAILS_URL),
+
+                'responseError' => $this->get(self::SETTING_RESPONSE_ERROR) ?: 'error',
+                'responseCode' => $this->get(self::SETTING_RESPONSE_CODE) ?: null,
+                'responseResourceOwnerId' => $this->get(self::SETTING_RESPONSE_RESOURCE_OWNER_ID) ?: 'id',
+            ]
+        );
+    }
+
+    /**
+     * @throws IdentityProviderException
+     *
+     * @return array user information, as returned by api_get_user_info(userId)
+     *
+     * @var AccessToken
+     * @var GenericProvider
+     */
+    public function getUserInfo($provider, $accessToken)
+    {
+        $url = $provider->getResourceOwnerDetailsUrl($accessToken);
+        $request = $provider->getAuthenticatedRequest($provider::METHOD_GET, $url, $accessToken);
+        $response = $provider->getParsedResponse($request);
+        if (false === is_array($response)) {
+            throw new UnexpectedValueException(get_lang('invalid_json_received_from_provider'));
+        }
+        $resourceOwnerId = $this->getValueByKey(
+            $response,
+            $this->get(self::SETTING_RESPONSE_RESOURCE_OWNER_ID)
+        );
+        if (empty($resourceOwnerId)) {
+            throw new RuntimeException(get_lang('wrong_response_resource_owner_id'));
```

Comment:
Variable inside get_lang() should be in PascalCase Example: 

`get_lang('WrongResponseResourceOwnerId');`

But it seems that this variable it's a plugin variable so, it should be 

`$this->get_lang('WrongResponseResourceOwnerId')`